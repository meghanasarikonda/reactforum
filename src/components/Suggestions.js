import React from 'react';
import PropTypes from 'prop-types';

const Suggestions = ({results, click}) => {

  const options = results.map(({id, username}) => (
    <div key={id}>
      <a href="0" data-id={id} key={id} onClick={click}>
        {username}
      </a>
    </div>
  ))
  console.log(options, 'options')
  return <div>{options}</div>
}

Suggestions.propTypes = {
  results: PropTypes.array,
  click: PropTypes.func
}

export default Suggestions;