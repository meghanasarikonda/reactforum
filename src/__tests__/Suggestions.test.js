import React from 'react';
import Enzyme, { shallow, mount, render } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';
import Suggestions from '../components/Suggestions';

function setup() {
  const myMock = jest.fn();
  myMock
    .mockReturnValueOnce('called once')
    .mockReturnValueOnce('called twice')

  const props = {
    results: [{username: 'Meghana', id: 1}],
    click: myMock
  }

  const enzymeWrapper = shallow(<Suggestions {...props} />)

  return {
    props,
    enzymeWrapper
  }
}

describe('Suggestions', () => {

  it('should render correctly', () => {
    const { enzymeWrapper } = setup()
    expect(shallowToJson(enzymeWrapper)).toMatchSnapshot();
  });

  it('should have 1 children', () => {
    const { enzymeWrapper } = setup()
    expect(enzymeWrapper.children().length).toBe(1);
  });

  it('onClick handler should call click prop', () => {
    const { enzymeWrapper, props } = setup()
    enzymeWrapper.find('div').simulate('click')
    expect(props.click.mock.calls.length).toBe(1);
    expect(props.click.mock.results[0].value).toBe('called once');
  });

  it('should render username', () => {
    const { enzymeWrapper, props } = setup()
    expect(enzymeWrapper.find('div').text()).toBe(props.results[0]['username'])
  });

  it('should contain div', () => {
    const { enzymeWrapper } = setup()
    expect(enzymeWrapper.exists('div')).toBe(true)
  });

  it('should return ul', () => {
    const { enzymeWrapper } = setup()
    expect(enzymeWrapper.find('ul').length).toEqual(1)
  });

})