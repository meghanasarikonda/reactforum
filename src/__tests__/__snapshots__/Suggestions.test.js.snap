// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Suggestions should render correctly 1`] = `
<span>
  <div
    key="1"
  >
    <a
      data-id={1}
      href="0"
      key="1"
      onClick={[MockFunction]}
    >
      Meghana
    </a>
  </div>
</span>
`;
